code/game/objects/items/weapons/stunbaton.dm

```diff
@@ -107,57 +107,73 @@
 		return
 	return ..()
 
-/obj/item/weapon/melee/baton/apply_hit_effect(mob/living/target, mob/living/user, var/hit_zone)
```

Comment:
I believe that the zone is checked later, also, I tested and it works as intended; if the baton is offline, no stun will be applied.